### PR TITLE
Add button to clone a rule in RuleTable

### LINKED ISSUE #1150: Add UI tool to clone a rule in RuleTable
### Description:
A tool to clone an existing rule in the `RuleTable` would be a nice improvement. Users can start new rules on existing ones. This can be important in case of having complex symbolizers and/or filters.

Suggestion: Another button `Clone Rule` in the bottom bar of the `RuleTable` UI, which clones all selected rows (by checkbox).

### PR DESCRIPTION
**Pull request by chrismayer**
This adds a tool to clone existing rules in the `RuleTable`, so users can start new rules based on existing ones.

grafik

Closes #1150

### CODE REVIEW

File: src/Component/Style/Style.tsx [hunk 1]
```diff
@@ -190,6 +191,35 @@ export class Style extends React.Component<StyleProps, StyleState> {
...
+    } = this.state;
+    const styleClone = _cloneDeep(style);
+
+    // create rules to clone
+    let newRules: GsRule[] = [];
+    styleClone.rules.forEach((rule: GsRule, index: number) => {
+      if (selectedRowKeys.includes(index)) {
+        let ruleClone = _cloneDeep(rule);
+        // TODO We need to ensure that rule names are unique
+        const randomId = Math.floor(Math.random() * 10000);
```
Comment (hwbllmnn) [inline src/Component/Style/Style.tsx]: Huh ;-) Maybe use https://github.com/kelektiv/node-uuid for this?

Comment (marcjansen) [reply src/Component/Style/Style.tsx]: Agreed we need to fix this. But in a separate PR, because we had this discussion elsewhere? Do you agree?

Comment (jansule) [reply src/Component/Style/Style.tsx]: Maybe lodash's [uniqueId()](https://lodash.com/docs/#uniqueId) can be helpful here. We also already use it in this project, as far as I know.

File: src/locale/es_ES.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ export default {
     },
     GsStyle: {
         addRuleBtnText: 'Añadir regla',
+        cloneRulesBtnText: 'Duplicadas reglas',
```
Comment (jansule) [inline src/locale/es_ES.ts]: Very minor: I think it should be `Duplicar reglas`.

Comment (chrismayer) [reply src/locale/es_ES.ts]: OK, my Spanish is a bit rusty and this is what an online translator gave me :laughing:

**Comment by chrismayer** [comment]
Thanks for you feedback guys! 

The random ID generation was taken from elsewhere in the code, so at least it is harmonized with other code places. I totally agree, that we should introduce something more robust in a separate PR. We should be aware, that there are several places in the GeoStyler code where this has to be changed.

Once Travis is green I'll merge this.

**Pull request merged**